manager/listen_linux.go

```diff
@@ -0,0 +1,30 @@
+package manager
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+// getDefaultListenIP on Linux uses netlink to get the default listen address by
+// asking the kernel what the route is to a public server.
+// This method is similar to what `ip route get <pubic server>` does
+func getDefaultListenIP() (string, error) {
+	routes, err := netlink.RouteGet(net.ParseIP("8.8.8.8"))
+	if err != nil {
+		var err2 error
+		// maybe it's v6 only
+		routes, err2 = netlink.RouteGet(net.ParseIP("2001:4860:4860:0:0:0:0"))
+		if err2 != nil {
+			// return the original err instead of err2
+			return "", err
+		}
+	}
+	for _, r := range routes {
+		// grab the first one
+		return r.Src.String(), nil
```

Comment:
You might get the link-local address as route source. I think you should include this check when going through the returned routes:
```
if r.Src.IsGlobalUnicast() {
   return r.Src.String(), nil
}
```

Route o/p example:
```
IPv6 route: [{Ifindex: 1 Dst: 2001:4860:4860::1/128 Src: fe80::eab1:fcff:fef5:118 Gw: <nil> Flags: [] Table: 0}]
```